src/lexer/lexer.cpp

```diff
@@ -2,35 +2,33 @@
 
 namespace jucc {
 
-int GetToken(FILE *fp) {
+int Lexer::GetToken(FILE *fp) {
   static int lastChar = ' ';
 
   while (lastChar != EOF && isspace(lastChar)) lastChar = getc(fp);
 
   // check for identifier and literal tokens
-
   // RE : [a-zA-Z][0-9a-zA-Z]
-
   if (isalpha(lastChar)) {
-    identifier_string = lastChar;
-    while (lastChar != EOF && isalnum(lastChar = getc(fp))) identifier_string += lastChar;
-    if (identifier_string == "int")
+    identifier_string_ = lastChar;
+    while (lastChar != EOF && isalnum(lastChar = getc(fp))) identifier_string_ += lastChar;
+    if (identifier_string_ == "int")
       return TOK_INT;
-    else if (identifier_string == "float")
+    else if (identifier_string_ == "float")
       return TOK_FLOAT;
-    else if (identifier_string == "void")
+    else if (identifier_string_ == "void")
       return TOK_VOID;
 
     // conditionals
-    else if (identifier_string == "if")
+    else if (identifier_string_ == "if")
       return TOK_IF;
-    else if (identifier_string == "else")
+    else if (identifier_string_ == "else")
       return TOK_ELSE;
 
     // cout , cin
-    else if (identifier_string == "cout")
+    else if (identifier_string_ == "cout")
       return TOK_COUT;
-    else if (identifier_string == "cin")
+    else if (identifier_string_ == "cin")
       return TOK_CIN;
 
     else
```

Comment:
line 40-to 41 (ref)[https://github.com/TheSYNcoder/JuCC/pull/1/files#diff-b38e901c69aa2066cc616dd83e46a27dbc89501a3a7b948a98fe9d0dfa16f61bL42-L43]

Instead of string, go with `std::stringstream`. Performance Improvements

Reply:
We will be already using fstream in the next push, so duplicating streams will not be efficient.